### PR TITLE
Friendly challenge alert

### PR DESCRIPTION
**Pull request by CLJunge**
Implements #3541

Translations for the added checkbox are missing, not sure about that.

### CODE REVIEW

File: HearthWatcher/FriendlyChallengeDialogVisibilityWatcher.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
+﻿using HearthWatcher.EventArgs;
+using HearthWatcher.Providers;
+using System;
+using System.Threading.Tasks;
+
+namespace HearthWatcher
+{
+	public class FriendlyChallengeDialogVisibilityWatcher
```
Comment (azeier) [inline HearthWatcher/FriendlyChallengeDialogVisibilityWatcher.cs]: Can we call this just "FriendlyChallengeWatcher"? :P

File: Hearthstone Deck Tracker/Hearthstone/Watchers.cs [hunk 2]
```diff
@@ -20,18 +20,21 @@ static Watchers()
 			PackWatcher.NewPackEventHandler += (sender, args) => PackUploader.UploadPack(args.PackId, args.Cards);
 			DungeonRunWatcher.DungeonRunMatchStarted += DeckManager.DungeonRunMatchStarted;
 			DungeonRunWatcher.DungeonInfoChanged += DeckManager.UpdateDungeonRunDeck;
+			FriendlyChallengeDialogVisibilityWatcher.OnDialogVisibilityChanged += (sender, args) => { if (args.DialogVisible && Config.Instance.FlashHsOnFriendlyChallenge) User32.FlashHs(); };
```
Comment (azeier) [inline Hearthstone Deck Tracker/Hearthstone/Watchers.cs]: No one line if statements please, this being multi line or a method is fine.

File: Hearthstone Deck Tracker/LogReader/Handlers/LoadingScreenHandler.cs [hunk 3]
```diff
@@ -75,6 +75,13 @@ public void Handle(LogLine logLine, IHsGameState gameState, IGame game)
 				else
 					Watchers.DungeonRunWatcher.Stop();
 
+				if (game.CurrentMode == Mode.HUB || game.CurrentMode == Mode.TOURNAMENT || game.CurrentMode == Mode.ADVENTURE
```
Comment (azeier) [inline Hearthstone Deck Tracker/LogReader/Handlers/LoadingScreenHandler.cs]: I feel like this should be moved into a prop (something like "MenuModes").

File: Hearthstone Deck Tracker/LogReader/Handlers/LoadingScreenHandler.cs [hunk 4]
```diff
@@ -75,6 +75,13 @@ public void Handle(LogLine logLine, IHsGameState gameState, IGame game)
...
+					|| game.CurrentMode == Mode.TAVERN_BRAWL || game.CurrentMode == Mode.DRAFT || game.CurrentMode == Mode.PACKOPENING
+					|| game.CurrentMode == Mode.COLLECTIONMANAGER)
+					Watchers.FriendlyChallengeDialogVisibilityWatcher.Run();
```
Comment (azeier) [inline Hearthstone Deck Tracker/LogReader/Handlers/LoadingScreenHandler.cs]: Run (and Stop?) should only be executed if the config setting is enabled.

File: Hearthstone Deck Tracker/Hearthstone/Watchers.cs [hunk 5]
```diff
@@ -52,4 +55,9 @@ public class HearthMirrorArenaProvider : IArenaProvider
 		public ArenaInfo GetArenaInfo() => DeckImporter.FromArena(false);
 		public HearthMirror.Objects.Card[] GetDraftChoices() => Reflection.GetArenaDraftChoices()?.ToArray();
 	}
+
+	public class HearthMirrorFriendlyChallengeDialogVisibilityProvider : IDialogVisibilityProvider
```
Comment (azeier) [inline Hearthstone Deck Tracker/Hearthstone/Watchers.cs]: We can probably shorten this to at least `HearthMirrorFriendlyChallengeProvider ` too.

File: HearthWatcher/FriendlyChallengeDialogVisibilityWatcher.cs (hunk 1 above)
Comment (CLJunge) [reply HearthWatcher/FriendlyChallengeDialogVisibilityWatcher.cs]: Agreed :)

File: Hearthstone Deck Tracker/Hearthstone/Watchers.cs (hunk 2 above)
Comment (CLJunge) [reply Hearthstone Deck Tracker/Hearthstone/Watchers.cs]: I'd like to create a method for this, not sure where to put it though.

Comment (azeier) [reply Hearthstone Deck Tracker/Hearthstone/Watchers.cs]: You can just add a static method in the Watchers class.

File: Hearthstone Deck Tracker/LogReader/Handlers/LoadingScreenHandler.cs (hunk 3 above)
Comment (CLJunge) [reply Hearthstone Deck Tracker/LogReader/Handlers/LoadingScreenHandler.cs]: Do you mean something like "PlayerCanBeChallenged"?

Comment (azeier) [reply Hearthstone Deck Tracker/LogReader/Handlers/LoadingScreenHandler.cs]: Something like that, yes.

**Comment by azeier** [comment]
Manually merged: https://github.com/HearthSim/Hearthstone-Deck-Tracker/compare/5706459bb8cb...9d9c74646aa5.

- [Fixed loading the checkbox state](https://github.com/HearthSim/Hearthstone-Deck-Tracker/compare/5706459bb8cb...9d9c74646aa5#diff-b187e6d8182acd6e438e7f97b5e033d8R49)
- Removed spaces between if/while and the parentheses.
